### PR TITLE
Make credentials notification run function testable

### PR DESCRIPTION
**Pull request by Nirvikalpa108**
## What does this change?
This PR refactors the run function in order to make it testable.

### CODE REVIEW

File: hq/app/schedule/CredentialsReportJob.scala [hunk 1]
```diff
@@ -21,22 +21,30 @@ class CredentialsReportJob(enabled: Boolean, cacheService: CacheService)(executi
...
-          logger.info(s"Completed scheduled job: $description")
+          createNotification(Account(awsAccount.id), message)
         }
       }
+      someValue.foreach{ result: Either[FailedAttempt, Notification] =>
+        result match {
+          case Left(value) =>
+            logger.info(s"check this issue out security devs! $value")
+          case Right(email) =>
+            send(email, topicArn, ???)(executionContext)
```
Comment (philmcmahon) [inline hq/app/schedule/CredentialsReportJob.scala:37]: what is the ??? for?

Comment (Nirvikalpa108) [reply hq/app/schedule/CredentialsReportJob.scala:37]: It's a placeholder for the AWS SNS client, which I will add in a subsequent PR. Sorry, I should have noted this PR is a WIP as I try to practice opening smaller PRs!

**Comment by philmcmahon** [comment]
I don't see how this makes the run function testable as it's still returning `Unit` - how do you intend to test it?

File: hq/app/schedule/CredentialsReportJob.scala [hunk 2]
```diff
@@ -21,22 +21,30 @@ class CredentialsReportJob(enabled: Boolean, cacheService: CacheService)(executi
     } else {
       logger.info(s"Running scheduled job: $description")
 
-      getCredentialsReport().foreach{ case (awsAccount, result) =>
-        result.foreach{ credsReport =>
+      val someValue: List[Either[FailedAttempt, Notification]] = getCredentialsReport().toList.map{ case (awsAccount, eFCreds) =>
```
Comment (Nirvikalpa108) [inline hq/app/schedule/CredentialsReportJob.scala:24]: I'm planning on testing this part of the code which has been refactored out of the foreach.

**Review by philmcmahon — APPROVED** [review]

**Pull request merged**